### PR TITLE
Add new validator to check rust audit is enforced

### PR DESCRIPTION
**Pull request by Firstyear**
After working with the security team, we decided that improving the standard of rust packaging will assist us with the application of and reaction to security issues. This is an initial template of a source validation tool that allows checking and verifying traits of packages that depend on rust.

### CODE REVIEW

**Comment by dirkmueller** [comment]
I appreciate the clean python implementation! I am a bit concerned as this is the first dependency on python for this service whether that is viable.

File: 80-rust-enforce-audit-capability [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+    # Build the set of buildrequries
+    buildrequires = [
+        l.split(":", 1)[1].strip()
+        for l in f_contents
+        if l.startswith("BuildRequires:")
+    ]
+
+    requires_rust = any([
+        x == "cargo" or x == "rust" or x == "cargo-packaging" or x == "rust+cargo"
```
Comment (dirkmueller) [inline 80-rust-enforce-audit-capability:68]:
```suggestion
        x in ('cargo', 'rust', 'cargo-packaging', 'rust+cargo'
```

**Review by dirkmueller — APPROVED** [review]

**Comment by Firstyear** [comment]
> I appreciate the clean python implementation! I am a bit concerned as this is the first dependency on python for this service whether that is viable.

OSC depends on python though? So this isn't really adding a new dep?

**Comment by coolo** [comment]
The obs services run within the source service server too though - but quite some of them already require python, so no worries IMO.  But due to your usage of /usr/bin/env I think the package requires will be wrong.

**Comment by Firstyear** [comment]
What would be better than bin env here?

**Comment by coolo** [comment]
There is no better or worse - but if you use env, you need to add a manual require (the discussion to make the auto provides handle it never ended anywhere). I'm afraid the spec file isn't maintained in git though, so it's more an action item for whoever accepts the MR.

**Comment by Firstyear** [comment]
So do I need to change anything in the PR before merge or not? If I change it to /usr/bin/python3 is that better? What is actionable here?

**Comment by coolo** [comment]
@bugfinder and @dirkmueller usually merge here, I was just commenting.

File: 80-rust-enforce-audit-capability [hunk 2]
```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
```
Comment (dirkmueller) [inline 80-rust-enforce-audit-capability:1]:
```suggestion
#!/usr/bin/python3
```

will help with package requires

**Comment by dirkmueller** [comment]
@Firstyear I'll try to catch bugfinder and then merge. if you have the time, please accept the shebang change that coolo pointed out.

**Comment by Firstyear** [comment]
Done, rebased to master and updated shebang

**Pull request merged**